include/deal.II/grid/tria_accessor.templates.h

```diff
@@ -2040,6 +2040,141 @@ TriaAccessor<structdim, dim, spacedim>::diameter () const
 
 
 
+namespace
+{
+  /**
+   * For a given center/radius, see if any points are outside.
+   * If so, enlarge the ball and return true;
+   * Otherwise, return false;
+   */
+  template <int structdim, int dim, int spacedim>
+  inline
+  bool
+  enlarge_sphere(std::array<bool,GeometryInfo<structdim>::vertices_per_cell> &vertices_within_ball,
+                 const TriaAccessor<structdim, dim, spacedim> &cell,
+                 Point<spacedim> &center,
+                 double &radius)
+  {
+    for (unsigned int v = 0; v < GeometryInfo<structdim>::vertices_per_cell; ++v)
+      if (!vertices_within_ball(v))
+        {
+          const double dist = center.distance(cell->vertex(v));
+          if (dist < radius)
+            {
+              vertices_within_ball[v] = true;
+            }
+          else
+            {
+              // we found a vertex which is outside of the ball
+              // extend it (move center and change radius)
+              const Point<spacedim> pCV = center - cell->vertex(v);
+              radius = (dist + radius) * 0.5;
+              center = cell->vertex(v) + pCV * (radius / dist);
+              return true;
```

Comment:
Actually, now that I have moved this chunk of code into the current function we don't need to update the vertex masks. Also thinking of adding:
```
for (unsigned int v = 0; v < GeometryInfo<structdim>::vertices_per_cell; ++v)
  Assert (center.distance(this->vertex(v)) < radius + 100. *std::numeric_limits<double>::epsilon(),
          ExcInternalError());
```